xbmc/pvr/channels/PVRChannelGroup.cpp

```diff
@@ -503,6 +504,17 @@ bool CPVRChannelGroup::IsMissingChannelGroupMembersFromClient(int iClientId) con
                    iClientId) != m_failedClientsForChannelGroupMembers.end();
 }
 
+void CPVRChannelGroup::UpdateClientOrder()
+{
+  CSingleLock lock(m_critSection);
+
+  for (const auto& member : GetMembers())
+  {
+    member.channel->SetClientOrder(member.iOrder);
+    member.channel->GetEPG()->GetChannelData()->SetClientOrder(member.iOrder);
```

Comment:
Updating epg channeldata should be done in the implementation of `CPVRChannel::SetClientOrder`, following the other implementations, example: `CPVRChannel::SetChannelNumber`=> https://github.com/xbmc/xbmc/blob/master/xbmc/pvr/channels/PVRChannel.cpp#L652